lib/rubocop/cop/commissioner.rb

```diff
@@ -99,6 +104,25 @@ def invoke_custom_processing(cops_or_forces, processed_source)
         end
       end
 
+      # There are cops that require their own custom processing **after**
+      # the AST traversal. By performing the walk before invoking these
+      # custom processors, we allow these cops to build their own
+      # state during the primary AST traversal instead of performing their
+      # own AST traversals. Minimizing the number of walks is more efficient.
+      #
+      # If they define the #investigate_after_walk method, all input parameters
+      # passed to the commissioner will be passed to the cop too in order to do
+      # its own processing.
+      def invoke_custom_processing_after_walk(cops, processed_source)
```

Comment:
Maybe `postprocessing` is a better term here?

Reply:
After trying a few variations, I settled on `_post_walk_processing`. The name `post_processing` suggests that what's meaningful is that these hooks are invoked after the initial `processing` round, ie after `invoke_custom_processing`. But actually, what's important is that these hooks invoked after _the walk_.

Feel free to reopen if you think otherwise.